lib/stanford-mods/concerns/origin_info.rb

```diff
@@ -81,32 +69,39 @@ def imprint_display_str
         imprints.map(&:display_str).reject(&:empty?).join('; ')
       end
 
-      # remove Elements from NodeSet if they have a qualifier attribute of 'approximate' or 'questionable'
-      # @param [Nokogiri::XML::Element] node the date element
-      # @return [Boolean]
-      # @private
-      def is_approximate(node)
-        qualifier = node["qualifier"] if node.respond_to?('[]')
-        qualifier == 'approximate' || qualifier == 'questionable'
-      end
+      private
+
+      # The rules for which date to pick is a little complicated:
+      #
+      # 1) Examine the date elements of the provided fields.
+      # 2) Discard any we can't parse a year out of.
+      # 3) (if ignore_approximate is true, used only by exhibits for Feigenbaum), throw out any qualified dates (or ranges if either the start or end is qualified)
+      # 4) If that set of date elements has elements with a keyDate, prefer those.
+      # 5) If there were encoded dates, prefer those.
+      # 6) Choose the earliest date (or starting date of a range).
+      #
+      # Finally, format the date or range of an encoded date, or just pluck out the year from an unencoded one.
+      def earliest_prefered_date(fields: [:dateIssued, :dateCreated, :dateCaptured], ignore_approximate: false)
```

Comment:
Spelling of "preferred"
```suggestion
      def earliest_preferred_date(fields: [:dateIssued, :dateCreated, :dateCaptured], ignore_approximate: false)
```